pkg/kubelet/kubelet_pods.go

```diff
@@ -487,6 +500,20 @@ func managedHostsFileContent(hostIPs []string, hostName, hostDomainName string,
 	return buffer.Bytes()
 }
 
+// isIPv6Supported determines whether the host supports ipv6 by check the global ipv6 support sysctl.
+// We use it to determine whether the pod should support ipv6.
+// But there are going to be edge cases we don't get right,
+// e.g. the user may have IPv6 enabled in the host netns,
+// but for some reason disable IPv6 in the pod netns.
+// But because kubelet cannot actually know about the loopback interface of the pod,
+// we cannot do something completely correct,
+// so we can only do simple and basically correct things.
+func isIPv6Supported() (bool, error) {
+	// TODO This should be injected by the caller.
+	val, err := utilsysctl.New().GetSysctl(sysctlAllDisableIPv6)
```

Comment:
can this value be only initialized once and cached?

Also, previous comment was asking about the windows. How was it resolved?

Reply:
> can this value be only initialized once and cached?

I think this was dicussed in another comment but I can not find it, the sysctl may change , do we want to use always the first value?

> Also, previous comment was asking about the windows. How was it resolved?

This file has plenty of "linux only" thing ... but to be honest I completely missed the windows angle

Reply:
> Also, previous comment was asking about the windows. How was it resolved?

This is an area we haven't considered before. Before this PR, the `makeHostsMount` function worked correctly on both `Linux` and `Windows`. However, the current PR breaks this logic. If it's Windows, it will return an error, which will cause the container creation to fail.

But we don't have a good way to determine if IPv6 is enabled on `Windows`.

When the system is Windows, can we bypass the `isIPv6` check? It will always be true or false.

Reply:
/lgtm cancel
/cc @jsturtevant 
/sig windows

for commenting on this

Reply:
I don't know that windows containers support ipv6 only.  There is a feature check for ipv6dualstack that could be used https://github.com/microsoft/hcsshim/blob/e55a82b1c962d666c68df8acb76bb311cd0a69db/hcn/hcnsupport.go#L32

@sbangari might be able to give more information about ipv6 only

Reply:
If we want this PR to proceed it sounds like we need new files, e.g. ipv6_linux.go and ipv6_nonlinux.go which hold the build-tagged implementations of this function (and the sysctl const)

Reply:
@SergeyKanzhelev  support for windows or other OS has been added, could you take a  look at it?